indico/modules/categories/templates/display/category_list.html

```diff
@@ -1,20 +1,27 @@
 {% macro category_list(parent_category) %}
     {% if parent_category.children %}
-        {% set category_title = parent_category.title %}
-        <h2 id="category-list" data-bypass-target="{% trans %}Skip to category list{% endtrans %}">
-            {%- trans %}Categories in {{ category_title }}{% endtrans -%}
-        </h2>
-        <ul class="category-list" data-subcat-info-url="{{ url_for('.subcat_info', parent_category) }}">
-            {% for category in parent_category.children %}
-                <li>
-                    <span class="protection {{ 'icon-shield' if category.is_self_protected }}"></span>
-                    <span class="number-events" id="category-event-count-{{ category.id }}"></span>
-                    <a href="{{ url_for('.display', category) }}" class="name">{{ category.title }}</a>
-                </li>
-            {% endfor %}
-        </ul>
-        <script>
-            setupCategoryDisplaySubcatList();
-        </script>
+        <ind-category-list subcat-info-url="{{ url_for('.subcat_info', parent_category) }}">
+            {% set category_title = parent_category.title %}
+            <h2 id="category-list" data-bypass-target="{% trans %}Skip to category list{% endtrans %}">
+                {%- trans %}Categories in {{ category_title }}{% endtrans -%}
+            </h2>
+            <ul class="category-list">
+                {% for category in parent_category.children %}
+                    <li id="category-{{ category.id }}">
+                        <a href="{{ url_for('.display', category) }}" class="name">
+                            {% if category.is_self_protected %}
+                                <span class="protected" aria-hidden="true"></span>
+                                <span class="protected-label">{% trans %}Protected category{% endtrans %}</span>
+                            {% else %}
+                                <span class="public" aria-hidden="true"></span>
```

Comment:
While this is just a CSS name, I'd prefer something else here than "public" since it can be either fully public or inheriting (from something that's either public or protected or inheriting itself). The idea of this icon here is to show you that there's special (typically more restrictive) protection on that category.

Reply:
So are you saying that it's basically just "not protected", and not necessarily "public"? I think I'll need to re-read it tomorrow. My brain's in something else atm. :)))

Reply:
Exactly! See https://learn.getindico.io/categories/protection/ (it's for categories but it's similar for all objects that use the protection system).

- Public = Everyone can access it. No other checks.
- Protected = Nobody except parent managers (those always get access) and those in the object's own ACL can access it
- Inheriting = Everyone who can access the parent object can access it and those in the object's own ACL can do so as well

That's also why the top-level category only allows setting Public or Protected, because it has no parent from which it could inherit something.

---

`is_self_protected` (coming from `ProtectionMixin`) checks the `protection_mode` setting stored on the object, while `is_protected` checks either the object itself or the parent, depending on the inheritance.

(and for odd legacy reasons I don't fully remember even though it was me (re-)implementing most of this 8 years ago, `is_public` behaves like `is_self_public` (which does not exist) 🙈 )

Reply:
No worries. We'll change that to `not-protected` or something. 

Btw, not sure if you noticed from my PRs but I don't treat classes as "CSS names". For me, the selector language is a query language, similar to SQL. So names always come from HTML which is being queried. By extension, the names in HTML should really reflect the content. (Except, of course, when I misunderstand something, haha.)